fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/storage/Storage.java

```diff
@@ -117,6 +117,20 @@ default boolean supportsExtraction() {
 	 */
 	Iterator<StorageView<T>> iterator(Transaction transaction);
 
+	/**
+	 * Iterate through the contents of this storage, for the scope of the passed transaction.
+	 * This function follows the semantics of {@link #iterator}, but returns an {@code Iterable} for use in {@code for} loops.
+	 *
+	 * @param transaction The transaction to which the scope of the returned iterator is tied.
+	 * @return An iterator over the contents of this storage.
+	 * @throws IllegalStateException If this storage is already exposing storage views.
+	 * @see #iterator
+	 */
+	default Iterable<StorageView<T>> iterable(Transaction transaction) {
+		Iterator<StorageView<T>> iterator = iterator(transaction);
+		return () -> iterator;
```

Comment:
```suggestion
		return () -> this.iterator(transaction);
```

Reply:
This is intended as iteration has mutually exclusive semantics, and I want `iterable` to acquire the "mutex" immediately.